vercel/resource_deployment_model.go

```diff
@@ -20,6 +20,13 @@ type ProjectSettings struct {
 	RootDirectory   types.String `tfsdk:"root_directory"`
 }
 
+// GitSource represents the Git integration source for a deployment.
+type GitSource struct {
+	RepoId types.String `tfsdk:"repo_id"`
+	Ref    types.String `tfsdk:"ref"`
+	Type   types.String `tfsdk:"type"`
+}
+
```

Comment:
I think we'd also want to add tests / documentation for this.

Reply:
working on these